modules/gcp/vendor-access/main.tf

```diff
@@ -31,23 +22,86 @@ resource "google_project_service" "gcp_apis" {
   service            = local.google_services[count.index]
 }
 
+locals {
+  is_impersonation_enabled = var.streamnative_org_id != ""
+  streamnative_gsa         = concat(var.streamnative_vendor_access_gsa, var.streamnative_support_access_gsa)
+}
+
+# Grant permissions directly to the StreamNative Cloud service account
+locals {
+  iam_bindings = local.is_impersonation_enabled ? [] : flatten([
+    for role in var.roles : [
+      for gsa in local.streamnative_gsa : {
+        role : role,
+        member : format("serviceAccount:%s", gsa),
+      }
+    ]
+  ])
+}
+
 resource "google_project_iam_member" "sn_access" {
-  for_each = {
-    for index, binding in local.iam_bindings :
-    index => binding
-  }
+  count = length(local.iam_bindings)
   project    = var.project
-  role       = each.value.role
-  member     = each.value.member
+  role       = local.iam_bindings[count.index].role
+  member     = local.iam_bindings[count.index].member
   depends_on = [google_project_service.gcp_apis]
 }
 
+# Grant permissions to the project service account that will be impersonated by StreamNative Cloud service account
+locals {
+  streamnative_bootstrap_gsa_name = format("snbootstrap-%s", var.streamnative_org_id)
```

Comment:
Suggestion: use `sncloud-bootstrap-%s` as prefix for consistent naming

Reply:
The max length for GSA name is 30 characters. If the organization ID length exceeds 12 characters, the GSA name will exceed the limit. So we should to keep the prefix as short as possible

Reply:
I see. Should we restrict the length of `var.streamnative_org_id` and do validation?

Reply:
Definitely yes

Reply:
added